### PR TITLE
Grave page

### PR DESCRIPTION
**Pull request by democrazyfi**

### CODE REVIEW

File: frontend/components/GravePageAssets.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { getGraveVaultFor } from '@/utils/universalProfile';
+import { Text } from '@chakra-ui/react';
+
+export default function GravePageAssets({ account }: { account: string }) {
+  const [graveVault, setGraveVault] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  useEffect(() => {
+    if (!graveVault) {
+      getGraveVaultFor(account)
```
[the remark this answers is not in the document]
Comment (democrazyfi) [reply frontend/components/GravePageAssets.tsx:13]: yeah wouldn't `if (!graveVault)` acheive that?

**Comment by tehnalogos** [comment]
@democrazyfi you some merge conflicts

File: frontend/components/GravePageAssets.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ledespana) [reply frontend/components/GravePageAssets.tsx:13]: not sure, I've never really understood that rule to be honest

File: frontend/utils/universalProfile.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  const graveForwarder = new ethers.Contract(
+    constants.UNIVERSAL_GRAVE_FORWARDER,
+    LSP1GraveForwader.abi,
+    provider
+  );
+  const graveYardAddress = await graveForwarder
+    .connect(signer)
+    .graveVaults(account);
+  return graveYardAddress === constants.ZERO_ADDRESS ? null : graveYardAddress;
```
[the remark this answers is not in the document]
Comment (Ledespana) [reply frontend/utils/universalProfile.ts:19]: If we do that we may have to change these:

- https://github.com/yearone-io/universal-grave/blob/9ae02efd16f61c2c11f7c215e2bcaa57a5c92d46/frontend/components/JoinGraveBtn.tsx#L185

- https://github.com/yearone-io/universal-grave/blob/9ae02efd16f61c2c11f7c215e2bcaa57a5c92d46/frontend/components/LSPAssets.tsx#L33

- https://github.com/yearone-io/universal-grave/blob/9ae02efd16f61c2c11f7c215e2bcaa57a5c92d46/frontend/components/MyGrave.tsx#L54

-https://github.com/yearone-io/universal-grave/blob/9ae02efd16f61c2c11f7c215e2bcaa57a5c92d46/frontend/components/wallet/WalletProvider.tsx#L31

-https://github.com/yearone-io/universal-grave/blob/9ae02efd16f61c2c11f7c215e2bcaa57a5c92d46/frontend/components/wallet/WalletProvider.tsx#L77

- https://github.com/yearone-io/universal-grave/blob/9ae02efd16f61c2c11f7c215e2bcaa57a5c92d46/frontend/components/wallet/WalletProvider.tsx#L121

[the remark this answers is not in the document]
Comment (democrazyfi) [reply frontend/utils/universalProfile.ts]: Sure, pushed - I think it simplifies things a bit

File: frontend/components/GravePageAssets.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tehnalogos) [reply frontend/components/GravePageAssets.tsx:13]: yep this only fires if account changes once.

**Review by tehnalogos — APPROVED** [review]

**Pull request merged**